### ISSUE TITLE
SPManagedMetadataServiceAppDefault: Update resource to allow the configuration of the defaults per Service App Proxy Group

### ISSUE THREAD

**Issue by ykuijs** [PowerShell/SharePointDsc#1136]
#### Details of the scenario you tried and the problem that is occurring
A support case for a customer revealed that the default MMS service app setting actually can be configured for one default per Service Application Proxy Group. The current version of the SPManagedMetadataServiceAppDefault resource only allows one default per farm. This resources needs updating to allow the configuration of one per proxy group. This would mean a Breaking Change, so can only be released with v4.0.

#### Suggested solution to the issue
Update resources to allow the configuration of one default per proxy group

#### The DSC configuration that is used to reproduce the issue (as detailed as possible)
```powershell
SPManagedMetaDataServiceAppDefault ManagedMetadataServiceAppDefault
{
    ServiceAppProxyGroup           = "ProxyGroupName"
    DefaultSiteCollectionProxyName = "Managed Metadata Service Application Proxy"
    DefaultKeywordProxyName        = "Managed Metadata Service Application Proxy"
    PsDscRunAsCredential           = $SetupAccount
}
```

#### The operating system the target node is running
Any

#### Version of SharePoint that is used (e.g. SharePoint 2016)
Any

#### Version and build of PowerShell the target node is running
v5.1

#### Version of the DSC module that was used ('dev' if using current dev branch)
dev

**Comment by andikrueger** [comment]
Good catch. Totally missed this scenario in the initial implementation of the resource.

**Comment by ykuijs** [comment]
Closed via PR #1137
